analysers/trials/IO.py

```diff
@@ -0,0 +1,70 @@
+import sys
+import json
+import io
+import gzip
+import uuid
+
+from pyspark_cassandra import CassandraSparkContext
+from pyspark import SparkConf
+
+# Takes arguments: Spark master, Cassandra host, Minio host, path of the files
+sparkMaster = sys.argv[1]
+cassandraHost = sys.argv[2]
+trialID = sys.argv[3]
+containerID = sys.argv[4]
+experimentID = trialID.split("_")[0]
+cassandraKeyspace = "benchflow"
+srcTable = "io_data"
+destTable = "trial_IO"
+
+# Set configuration for spark context
+conf = SparkConf() \
+    .setAppName("I/O analyser") \
+    .setMaster(sparkMaster) \
+    .set("spark.cassandra.connection.host", cassandraHost)
+sc = CassandraSparkContext(conf=conf)
+
+def f(a, b):
+    if a>b:
+        return a
+    else:
+        return b
+
+data = sc.cassandraTable(cassandraKeyspace, srcTable)\
+        .select("device") \
+        .where("trial_id=? AND experiment_id=? AND container_id=?", trialID, experimentID, containerID) \
+        .collect()
+
+devices = {}
+for e in data:
+    dev = e["device"]
+    devices[dev] = 0
+
+queries = []
+
+for d in devices.keys():
```

Comment:
This can be a query where we get all the data for all the devices in a given experiment's trial with `max(read,write,total)`. Since all the counters are incremented it should return only one row, or if more we can do a `SELECT DISTINCT` or just use the first one since they are gonna be the same data. Then we do post processing on it, summing up the `read`, the `write` and the `total` among all the devices. For now I would say this is an acceptable solution, but please open an issue about what I discussed above and map the issue to a reply of this comment.